sugar.go

```diff
@@ -212,25 +208,35 @@ func (s *SugaredLogger) sweetenFields(args []interface{}) []zapcore.Field {
 	if len(args) == 0 {
 		return nil
 	}
-	if len(args)%2 == 1 {
-		s.core.DPanic(_oddNumberErrMsg, Any("ignored", args[len(args)-1]))
-	}
 
-	fields := make([]zapcore.Field, len(args)/2)
-	for i := range fields {
-		keyIdx := 2 * i
-		val := args[keyIdx+1]
-		key, ok := args[keyIdx].(string)
+	// Allocate enough space for the common case; if users only pass structured
+	// fields, we'll need to grow the slice.
+	fields := make([]zapcore.Field, 0, len(args)/2)
```

Comment:
hmmm... I'd argue for using `len(args)` as the default; i.e. conservatively over-estimate, precisely so that if you start converting to "better" structured fields, you don't get a surprising perf/allocs "cliff" behavior.